### ISSUE TITLE
Uncaught (in promise) FlickingError: Animation is already playing. 

### ISSUE THREAD

**Issue by robjewell-ibotta** [naver/egjs-flicking#474]
## Description

Currently seeing this error when clicking either `flicking.prev();` or `flicking.next();` a lot and quickly from the browser.
```
Uncaught (in promise) FlickingError: Animation is already playing.
```

Is this a throttle issue? Does flicking have any throttle features or debounce features to prevent errors from the above relatively aggressive clicking behavior?

These are my dependencies:
```
 "@egjs/flicking": "^4.1.0",
 "@egjs/vue-flicking": "^4.1.0"
 ```

**Comment by WoodNeck** [comment]
Hello @robjewell-ibotta!
That's an expected behavior, like #473.
`prev/next/moveTo` should be resolved only when Flicking reaches the target panel.

You can easily suppress the errors like this:

```js
// Simple
flicking.next().catch(() => void 0);

// If you want to be more precise
flicking.next()
  .catch(err => {
    if (err instanceof FlickingError) return;
    throw err;
  });
```

**Comment by robjewell-ibotta** [comment]
Hi @WoodNeck , thank you for your response! I actually came up with a solution that I think prevents the error from happening:

On each click callback which call `prev()`, `next()` or `slideTo()`, I perform some logic that prevents the methods from being called rapidly. I will demonstrate what I mean with the callback that uses `next()`:

I have a method called, `next`, which is bound to a click event.:

```js
next() {
    this.nextClicked = true;
    if (!this.isSliding) {
      const flick = this.$refs.flick
      flick.next();
      this.nextClicked = false;
    }
  }
``` 

In the above method `next()` callback assigns boolean values to local state in the component. If the carousel is presently sliding, the method blocks the click event from sliding (` if (!this.isSliding) {`). If the carousel is not sliding, then the `next()` callback allows: `flick.next();` to run. 

In conjunction with the above click callback there is another callback which I bind to the event `@move-end`:

```js
onMoveEnd(event: any) {
    this.isSliding = false;

    if (this.nextClicked) {
      this.next();
    }
  }
```

If the carousel is ever in the state where a user is rapidly clicking the "next" button, then once the carousel has finished moving `this.nextClicked` will be in a boolean state true. If this is the case, then we execute the callback `this.next()`, which I describe in detail above. 

The above logic prevents the exception from ever being raised because moving events -- `next()`, `prev()`, `slideTo()` -- are only executed when their operation is complete and never at the same time.  Right now, the above approach is preventing the situation that causes the exception, which I think is better than only suppressing the error when it happens. 

What are your thoughts?

**Comment by WoodNeck** [comment]
@robjewell-ibotta 
It's okay to do that if you think it's safer to do that, but you should consider other causes of error like when the user input has interrupted the animation or the event's stop is called.

Rejecting the error may somehow seem absurd, but it can be also a chance of customization.
Like, if you want to add some additional effects, like shaking the arrow button, when the button is clicked while Flicking's moving you can simply do it like this:

```js
flicking.next()
  .catch(err => {
    if (err instanceOf FlickingError && err.code === ERROR_CODE.ANIMATION_ALREADY_PLAYING) {
      // Show some effects
    } else {
      return;
    }
  });
```

So basically, I think you can safely ignore the error with the multiple clicks or [other cases of errors](https://naver.github.io/egjs-flicking/docs/api/Flicking#next)
Also, there's a [Arrow](https://naver.github.io/egjs-flicking/Plugins#arrow) plugin that just released if you need it :)

**Comment by robskrob** [comment]
@WoodNeck thank you for the above explanation

**Comment by robjewell-ibotta** [comment]
Thank you @WoodNeck !
